### ISSUE TITLE
Server error - Create new email alias 

### ISSUE THREAD

**Issue by Nomarix** [simple-login/app#211]
Hi,

When I try to create on my self hosted solution a new Alias with my own letters/words, I always end up in:
"Server error, Looks like we are having some server issues..., We are notified and will look at this issue asap!". 
However Random Aliases and UUIDs works fine.

I tried to narrow down the issue but I get stucked as well I also didn't found a helpful log file.
Do you observered a similar issue before?
Where is the best place to start debugging?
Any hints which I should try or can check?

Thanks

**Comment by nguyenkims** [comment]
@Nomarix You can see the log in the webapp docker container

```bash
docker logs -f sl-app
```

Can you tell me what does the log say when this error happens?

You can also have a more detailed error logs if Sentry is enabled.

**Comment by jordanmq** [comment]
Hi all,
Not sure if I should open a separate issue but I have thee exact same behaviour with a self hosted instance.

Here is the output of `docker logs -f sl-app`

`2020-05-26 11:08:11,028 - sl - ERROR - 8 - server:397 - error_handler - 'NoneType' object has no attribute 'id'
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1832, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1818, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/local/lib/python3.7/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/code/app/dashboard/views/custom_alias.py", line 133, in custom_alias
    custom_domain_id = domain.id
AttributeError: 'NoneType' object has no attribute 'id'`

**Comment by Nomarix** [comment]
Hi nguyenkims,

thanks for the hint. Find below the output.

Additionally I tried to turn off (DISABLE_ALIAS_SUFFIX=1) but this is also not working.
Changed the param, did restart but no address adder is shown in the Create Alias page. 
Not sure if there is a direct correlation but seems to pointing in the same direction.

Thanks

... - sl - ERROR - 14 - server:337 - error_handler - 'NoneType' object has no attribute 'id'
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1832, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1818, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/local/lib/python3.7/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/code/app/dashboard/views/custom_alias.py", line 108, in custom_alias
    alias.custom_domain_id = domain.id
AttributeError: 'NoneType' object has no attribute 'id'

**Comment by SibrenVasse** [comment]
@nguyenkims

Managed to reproduce with 3.1.0.
alias_suffix.startswith("@") is always true, also for domains from OTHER_ALIAS_DOMAINS.
CustomDomain.get_by(domain=alias_domain) will then return None.

**Comment by nguyenkims** [comment]
@SibrenVasse thanks! This is indeed an issue with self-hosting as the custom domain table is empty. 

@Nomarix the issue has been fixed in 3.1.1, please upgrade to this version using this guide here https://github.com/simple-login/app/blob/master/docs/upgrade.md

**Comment by Nomarix** [comment]
@nguyenkims @SibrenVasse 👍 

Thank you, it is running fine now. 
Thanks as well for the quick fix and the fast released upgrade sequence 🥇

**Comment by nguyenkims** [comment]
@inonio there's this regression in 3.2.0 indeed. We just published 3.2.1 that fixes that issue. Could you please try to upgrade to 3.2.1? The upgrade guide is https://github.com/simple-login/app/blob/220c7bdf11b061a9a96344cd40e786a610de1087/docs/upgrade.md#L166

**Comment by LordChunk** [comment]
@nguyenkims the issue seems to still occur in version 3.2.1
```Log
'NoneType' object has no attribute 'id'
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1832, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1818, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/local/lib/python3.7/site-packages/flask_cors/decorator.py", line 128, in wrapped_function
    resp = make_response(f(*args, **kwargs))
  File "/code/app/api/base.py", line 32, in decorated
    return f(*args, **kwargs)
  File "/code/app/api/views/new_custom_alias.py", line 173, in new_custom_alias_v2
    custom_domain_id = domain.id
AttributeError: 'NoneType' object has no attribute 'id'
```

**Comment by LordChunk** [comment]
Interestingly, I do seem to be able to create custom email addresses from the dashboard, but not from the Chrome plugin.
![image](https://user-images.githubusercontent.com/23651869/84696965-98da3000-af4d-11ea-874e-fcb7cb0a4a58.png)

**Comment by nguyenkims** [comment]
@LordChunk there's a bug indeed when an alias is created from the mobile app or browser extension when `DISABLE_ALIAS_SUFFIX` is set. This bug has been fixed in https://github.com/simple-login/app/commit/1a22d0cf9b168b1461e5b9ab1df93118f4bc2e01 and is available in v3.2.2. 

Could you upgrade to this version and let me know if the issue persists? The upgrade guide is in https://github.com/simple-login/app/blob/3d0d42c8b3269e9922819cc04761ec7ffffe1838/docs/upgrade.md#L122-L121

**Comment by LordChunk** [comment]
@nguyenkims it works again. 😄
